source/apiVolontaria/volunteer/views.py

```diff
@@ -344,6 +344,20 @@ def get_queryset(self):
             queryset = queryset.filter(user__username=username)
         return queryset
 
+    def post(self, request, *args, **kwargs):
+        response = self.create(request, *args, **kwargs)
+
+        if response.status_code == 201:
+            obj_id = response.data['id']
+            obj = models.Participation.objects.get(pk=obj_id)
+
+        must_generate_email_invitation = request.data.get('send_ics', False)
+
+        if obj and must_generate_email_invitation is True:
+            obj.generate_email_invitation()
```

Comment:
I think we want to move this logic in the serializer to respect the logic of @francisjeanneau